src/test/ui/issues/issue-67552.stderr

```diff
@@ -10,11 +10,7 @@ note: `rec` defined here
 LL | / fn rec<T>(mut it: T)
 LL | | where
 LL | |     T: Iterator,
```

Comment:
I personally might even prefer to exclude where bounds from the function signature :thinking:

Reply:
so we only point at `fn rec<T>(mut it: T)` here

Reply:
I think it's better to display the `where` clause, so switching from `fn<T: Copy>() {}` to `fn<T>() where T: Copy {}` doesn't change the diagnostic output.

Reply:
I don't fully get what you mean here, wouldn't that change change the diagnostic output from `fn<T: Copy>()` to `fn<T>() where T: Copy` with this PR?

I think that diagnostics where where bounds are relevant mention the where bound itself, so I think that not mentioning them in the signature is worth it as it can remove noise.

Reply:
Sorry - I meant that if a user switches from an 'inline' `<T: Copy>` to a 'where-clause `where T: Copy`, they should still see the same amount of information in diagnostics.

Reply:
hmm, I am not sure how often these bounds help and are not explicitly references and put a fairly high value into single line error spans.

But that's just a personal preference and I am fine with landing this as is, so r=me. We may want to wait a day or two in case @estebank want's to look over this but otherwise I am satisfied